appsdk/sdk.go

```diff
@@ -511,35 +571,98 @@ func (sdk *AppFunctionsSDK) listenForConfigChanges() {
 			}
 
 			previousLogLevel := sdk.config.Writable.LogLevel
+			previousStoreForward := sdk.config.Writable.StoreAndForward
 
 			sdk.config.Writable = *actual
-			sdk.LoggingClient.SetLogLevel(sdk.config.Writable.LogLevel)
 			sdk.LoggingClient.Info("Writable configuration has been updated from Registry")
 
-			if previousLogLevel != sdk.config.Writable.LogLevel {
-				// Log level changed, not Pipeline, so skip updating the pipeline
-				continue
+			// Note: Changes occur one setting at a time so if setting not part of the pipeline,
+			//       then skip updating the pipeline
+			switch {
+			case previousLogLevel != sdk.config.Writable.LogLevel:
+				sdk.LoggingClient.SetLogLevel(sdk.config.Writable.LogLevel)
+				sdk.LoggingClient.Info(fmt.Sprintf("Logging level changed to %s", sdk.config.Writable.LogLevel))
+
+			case previousStoreForward.MaxRetryCount != sdk.config.Writable.StoreAndForward.MaxRetryCount:
+				if sdk.config.Writable.StoreAndForward.MaxRetryCount < 0 {
+					sdk.LoggingClient.Info(fmt.Sprintf("StoreAndForward MaxRetryCount can not be less than 0, defaulting to 1"))
```

Comment:
Kinda feel like this should be a warning, not Info

Reply:
Agreed.